Intermittent NPE at testCheckCastValueTypeOnInvalidQtype with -Xjit:count=0

Intermittent NPE at `TestCheckCastOnInvalidQtype.testCheckCastOnInvalidQtype` while running `ValueTypeTests` with `-Xjit:count=0`. It's hard to reproduce when enabling verbose log or trace log. 

Related to #9392

```
===============================================
Running test ValueTypeTests_4 ...
===============================================
ValueTypeTests_4 Start Time: Tue Jun 30 11:25:33 2020 Epoch Time (ms): 1593541533127
"/root/home/ahuo/src/openj9-openjdk-jdk11/build/linux-x86_64-normal-server-release/images/jdk/bin/java" -Xshareclasses:destroyAll; "/root/home/ahuo/src/openj9-openjdk-jdk11/build/linux-x86_64-normal-server-release/images/jdk/bin/java" -Xshareclasses:groupAccess,destroyAll; echo "cache cleanup done";
JVMSHRC005I No shared class caches available
JVMSHRC005I No shared class caches available
cache cleanup done
variation: -Xjit:count=0
JVM_OPTIONS:  -Xjit:count=0 
{ itercnt=1; \
mkdir -p "/root/home/ahuo/src/openj9-openjdk-jdk8/openj9/test/TKG/../TKG/test_output_15935415184246/ValueTypeTests_4"; \
cd "/root/home/ahuo/src/openj9-openjdk-jdk8/openj9/test/TKG/../TKG/test_output_15935415184246/ValueTypeTests_4"; \
"/root/home/ahuo/src/openj9-openjdk-jdk11/build/linux-x86_64-normal-server-release/images/jdk/bin/java"  -Xjit:count=0  \
	-Xverify:none \
	--add-opens java.base/jdk.internal.misc=ALL-UNNAMED \
	-XX:+EnableValhalla \
	-cp "/root/home/ahuo/src/openj9-openjdk-jdk8/openj9/test/TKG/../TKG/lib/asm-all.jar:/root/home/ahuo/src/openj9-openjdk-jdk8/openj9/test/TKG/../../jvmtest/TestConfig/resources:/root/home/ahuo/src/openj9-openjdk-jdk8/openj9/test/TKG/../TKG/lib/testng.jar:/root/home/ahuo/src/openj9-openjdk-jdk8/openj9/test/TKG/../TKG/lib/jcommander.jar:/root/home/ahuo/src/openj9-openjdk-jdk8/openj9/test/TKG/../../jvmtest/functional/Valhalla/ValhallaTests.jar" \
	org.testng.TestNG -d "/root/home/ahuo/src/openj9-openjdk-jdk8/openj9/test/TKG/../TKG/test_output_15935415184246/ValueTypeTests_4" "/root/home/ahuo/src/openj9-openjdk-jdk8/openj9/test/TKG/../../jvmtest/functional/Valhalla/testng.xml" -testnames ValueTypeTests \
	-groups level.sanity \
	-excludegroups d.*.linux_x86-64_cmprssptrs,d.*.arch.x86,d.*.os.linux,d.*.bits.64,d.*.generic-all; \
	if [ $? -eq 0 ] ; then echo ""; echo "ValueTypeTests_4""_PASSED"; echo ""; cd /root/home/ahuo/src/openj9-openjdk-jdk8/openj9/test/TKG/..;  else echo ""; echo "ValueTypeTests_4""_FAILED"; echo ""; fi; } 2>&1 | tee -a "/root/home/ahuo/src/openj9-openjdk-jdk8/openj9/test/TKG/../TKG/test_output_15935415184246/TestTargetResult";
[IncludeExcludeTestAnnotationTransformer] [INFO] exclude file is /root/home/ahuo/src/openj9-openjdk-jdk8/openj9/test/TKG/../TestConfig/resources/excludes/latest_exclude_Valhalla.txt
[IncludeExcludeTestAnnotationTransformer] [INFO] Unable to open file /root/home/ahuo/src/openj9-openjdk-jdk8/openj9/test/TKG/../TestConfig/resources/excludes/latest_exclude_Valhalla.txt, changed to read /root/home/ahuo/src/openj9-openjdk-jdk8/openj9/test/TKG/../TestConfig/resources/excludes/latest_exclude_base.txt
...
... TestNG 6.14.2 by Cédric Beust (cedric@beust.com)
...
PASSED: testCheckCastRefClassOnNull
PASSED: testCheckCastValueTypeOnInvalidLtype
PASSED: testCheckCastValueTypeOnNonNullType
PASSED: testCheckCastValueTypeOnNull
PASSED: testCreatePoint2D
...
...

FAILED: testCheckCastValueTypeOnInvalidQtype
org.testng.TestException: 
Expected exception of type class java.lang.NoClassDefFoundError but got java.lang.NullPointerException
	at org.testng.internal.ExpectedExceptionsHolder.wrongException(ExpectedExceptionsHolder.java:68)
	at org.testng.internal.Invoker.considerExceptions(Invoker.java:1129)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:612)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:716)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:988)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.testng.TestNG.privateMain(TestNG.java:1354)
	at org.testng.TestNG.main(TestNG.java:1323)
Caused by: java.lang.NullPointerException
	at TestCheckCastOnInvalidQtype.testCheckCastOnInvalidQtype(TestCheckCastOnInvalidQtype.java)
	at java.base/java.lang.invoke.DirectHandle.invokeExact_thunkArchetype_L(DirectHandle.java:289)
	at java.base/java.lang.invoke.FilterReturnHandle.invokeExact_thunkArchetype_X(FilterReturnHandle.java:59)
	at org.openj9.test.lworld.ValueTypeTests.testCheckCastValueTypeOnInvalidQtype(ValueTypeTests.java:2025)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:580)
	... 18 more
```

--- a/runtime/compiler/ilgen/Walker.cpp
+++ b/runtime/compiler/ilgen/Walker.cpp
@@ -2364,6 +2364,8 @@ TR_J9ByteCodeIlGenerator::genCheckCast()
  *
  * If the class specified in the bytecode is unresolved, this leaves out the
  * ResolveCHK since it has to be conditional on a non-null object.
+ * If the class specified in the bytecode is a value type, it has to be resolved
+ * unconditionally, regardless of whether the value is null.
  *
  * @param cpIndex The constant pool entry of the class given in the bytecode
  *
@@ -2376,12 +2378,16 @@ TR_J9ByteCodeIlGenerator::genCheckCast(int32_t cpIndex)
    if (TR::Compiler->om.areValueTypesEnabled() && TR::Compiler->cls.isClassRefValueType(comp(), method()->classOfMethod(), cpIndex))
       {
       TR::Node * objNode = _stack->top();
+
+      loadClassObject(cpIndex);
+
       TR::Node *passThruNode = TR::Node::create(TR::PassThrough, 1, objNode);
       genTreeTop(genNullCheck(passThruNode));
-      loadClassObject(cpIndex);
       }
    else
+      {
       loadClassObjectForTypeTest(cpIndex, TR_DisableAOTCheckCastInlining);
+      }
    genCheckCast();
    }
 

